src/NServiceBus.Core/Pipeline/StageForkConnector.cs

```diff
@@ -22,7 +22,7 @@ public Task Invoke(TFromContext context, Func<TToContext, Task> next)
 
             return Invoke(context, next, ctx =>
             {
-                var cache = context.Extensions.Get<IPipelineCache>();
+                var cache = ctx.Extensions.Get<IPipelineCache>();
```

Comment:
This is crucial to make the delegate cacheable. Without this we create a `Func `for each call.